components/org.wso2.carbon.deployment.automation/src/main/java/org/wso2/carbon/deployment/automation/DeploymentService.java

```diff
@@ -192,19 +183,11 @@ private DeploymentProvider getDeploymentProvider() throws DeploymentEnvironmentE
             throw new DeploymentEnvironmentException("Unable to identify the deployment platform.");
         }
 
-        if (platform.toLowerCase().equals(DEPLOYMENT_KUBERNETES)) {
+        if (platform.equalsIgnoreCase(DEPLOYMENT_KUBERNETES)) {
```

Comment:
Thanks for the feedback Dilan! We have added a null check before this statement therefore it might not through a NPE, anyway will change the order of this statement to suite the best practice.